### DISCUSSION TITLE
Simulating second order cross-terms

### DISCUSSION THREAD

**Discussion by subhradip-paul** [deepanshs/mrsimulator#388] [Q&A] [answered]
Recently we were investigating a situation where we would like to simulate a spectrum of spin 1/2 coupled to spin 3/2. This would involve a cross term between dipolar coupling and quadrupolar coupling as mentioned in the following paper:

_Eichele K, Wasylishen RE, Grossert JS, Olivieri AC. 1995. J. Phys. Chem. 99 (25): 10110–13_
I was trying to replicate the 9.4 T spectrum in Fig 2 of the aforementioned paper.

I understand that you need the *IxSz* part of the dipolar Hamiltonian as mentioned in the paper:
Menger EM, Veeman WS. 1982. Journal of Magnetic Resonance (1969). 46(2): 257–68

Is it possible to do in mrsimulator? In general, is it possible to just write the full Hamiltonian and diagonalise, but I was just wondering if it is possible to do in Mrsimulator. 

Thanks,
Paul

**Comment by deepanshs** [comment, marked as answer]
Hi Paul,

We have an upcoming release that include the second-order perturbation theory correction involving the quadrupolar interaction with 1) shielding, 2) weak J-coupling, and 3) weak dipolar coupling. For more information on mrsimulator theory and limitations, please refer to [J. Chem. Phys. 160, 234110 (2024)](https://doi.org/10.1063/5.0209887)

The version is not officially released but you may try out the release candidate, `pip install "mrsimulator==1.0.0rc1"`
There are few examples in the latest doc, such as, [Influence of 14N on 13C NMR MAS spectra of glycine](https://mrsimulator.readthedocs.io/en/latest/examples/1D_simulation%28crystalline%29/plot_13_quad_dipolar.html#sphx-glr-examples-1d-simulation-crystalline-plot-13-quad-dipolar-py).

Please let us know if you see any issue with the installation or simulation. Happy to assist.

---
> In general, is it possible to just write the full Hamiltonian and diagonalise, but I was just wondering if it is possible to do in Mrsimulator.

As of now, mrsimulator does not support simulation involving diagonalizing full Hamiltonian. All frequency corrections are base on the first and second-order perturbation in high field approximation limit.

-Best,
Deepansh

**Reply by subhradip-paul** [reply]
OK thank you very much for the help. I will install the release candidate and try it out.

thanks,
Paul
